common/src/main/java/com/tc/net/protocol/transport/ConnectionHealthCheckerImpl.java

```diff
@@ -91,22 +110,15 @@ public void start() {
   }
 
   @Override
-  public void stop() {
-    if (shutdown.attemptSet()) {
-      monitorThreadEngine.stop();
-      monitorThread.cancel();
-// don't bother to join the monitorThread here.  shutdown should take care of all the
-// threads in the thread group
+  public synchronized void stop() {
+    if (task != null && !task.isCancelled()) {
```

Comment:
This might be better as `!task.isDone()`.

Reply:
I don't want to re-issue the cancel if it has already been initiated.